src/sql/parts/objectExplorer/common/objectExplorerViewTreeShim.ts

```diff
@@ -119,7 +119,20 @@ export class OEShimService extends Disposable implements IOEShimService {
 	private treeNodeToITreeItem(viewId: string, node: TreeNode, parentNode: ITreeItem): ITreeItem {
 		let handle = generateUuid();
 		let nodePath = node.nodePath;
-		let newTreeItem = {
+		let icon: string;
+		if (node.iconType) {
+			icon = (typeof node.iconType === 'string') ? node.iconType : node.iconType.id;
+		} else {
+			icon = node.nodeTypeId;
```

Comment:
Is node.nodeTypeId guaranteed to be non-null?

Reply:
As far as I know yes. I guess the only risk here is calling `.toLowerCase()` on undefined. I can initialize it to a blank string to get around this.